From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 30 Apr 2026 12:00:00 +0200
Subject: [PATCH] dts: Fix pinctrl errors and LTDC probe for Decktrix board

Disable peripherals not present on Decktrix, remove LTDC parallel
RGB pinctrl (DSI output doesn't need it), and disable HDMI transmitter
to break LTDC dependency cycle. Fixes all "invalid function" and
"pin not registered" pinctrl errors at boot, and allows LTDC to
probe through the DSI path.

- Disable &ethernet0 (no PHY, fixes PI5 AF12/AF11)
- Delete LTDC pinctrl (DSI doesn't need parallel RGB pins, fixes PI8/PI10)
- Disable &sdmmc2 (WiFi uses different module, fixes PH4)
- Disable &usart2 (BT uses UART8 on Decktrix, fixes PI3/PI11)
- Disable HDMI transmitter (breaks LTDC deferred probe cycle)

---
 arch/arm/boot/dts/st/stm32mp157c-dk2-jadard.dts | 42 +++++-----------
 arch/arm/boot/dts/st/stm32mp15xx-dkx.dtsi       |  2 +-
 2 files changed, 11 insertions(+), 33 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp157c-dk2-jadard.dts b/arch/arm/boot/dts/st/stm32mp157c-dk2-jadard.dts
index ba87db888..3cca06d9c 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-dk2-jadard.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2-jadard.dts
@@ -45,6 +45,10 @@ &cryp1 {
 	status = "okay";
 };
 
+&ethernet0 {
+	status = "disabled";
+};
+
 &dsi {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -91,6 +95,9 @@ jadard@68 {
 };
 
 &ltdc {
+	/delete-property/ pinctrl-names;
+	/delete-property/ pinctrl-0;
+	/delete-property/ pinctrl-1;
 	status = "okay";
 
 	port {
@@ -114,44 +121,12 @@ rtc_lsco_pins_a: rtc-lsco-0 {
 	};
 };
 
-/* Wifi */
+/* Wifi - disabled, Decktrix uses u-blox MAYA-W166 */
 &sdmmc2 {
-	pinctrl-names = "default", "opendrain", "sleep";
-	pinctrl-0 = <&sdmmc2_b4_pins_a>;
-	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
-	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
-	non-removable;
-	cap-sdio-irq;
-	st,neg-edge;
-	bus-width = <4>;
-	vmmc-supply = <&v3v3>;
-	mmc-pwrseq = <&wifi_pwrseq>;
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-
-	brcmf: wifi@1 {
-		reg = <1>;
-		compatible = "brcm,bcm4329-fmac";
-		pinctrl-names = "default";
-		pinctrl-0 = <&rtc_lsco_pins_a>;
-	};
+	status = "disabled";
 };
 
-/* Bluetooth */
+/* Bluetooth - disabled, Decktrix uses UART8 */
 &usart2 {
-	pinctrl-names = "default", "sleep", "idle";
-	pinctrl-0 = <&usart2_pins_c>;
-	pinctrl-1 = <&usart2_sleep_pins_c>;
-	pinctrl-2 = <&usart2_idle_pins_c>;
-	uart-has-rtscts;
-	status = "okay";
-
-	bluetooth {
-		shutdown-gpios = <&gpioz 6 GPIO_ACTIVE_HIGH>;
-		compatible = "brcm,bcm43438-bt";
-		max-speed = <3000000>;
-		vbat-supply = <&v3v3>;
-		vddio-supply = <&v3v3>;
-	};
+	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/st/stm32mp15xx-dkx.dtsi b/arch/arm/boot/dts/st/stm32mp15xx-dkx.dtsi
index a3a45e026..71b7072e7 100644
--- a/arch/arm/boot/dts/st/stm32mp15xx-dkx.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15xx-dkx.dtsi
@@ -188,7 +188,7 @@ hdmi-transmitter@39 {
 		interrupts = <1 IRQ_TYPE_EDGE_FALLING>;
 		interrupt-parent = <&gpiog>;
 		#sound-dai-cells = <0>;
-		status = "okay";
+		status = "disabled";
 
 		ports {
 			#address-cells = <1>;
-- 
2.43.0

